### PR TITLE
"add tensorflow lstm"

### PR DESCRIPTION
**Pull request by dzhwinter**
fix https://github.com/PaddlePaddle/Paddle/issues/6129

### CODE REVIEW

File: tensorflow/understand_sentiment_lstm.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+            buf_size=FLAGS.batch_size * 10),
+        batch_size=FLAGS.batch_size)
+
+    data = tf.placeholder(tf.int64, shape=[None, FLAGS.seq_len])
+    label = tf.placeholder(tf.int64, shape=[None])
+    prediction = model(data, dict_dim)
+    cost = tf.nn.softmax_cross_entropy_with_logits(prediction, label)
+    avg_cost = tf.reduce_mean(cost)
+    adam_optimizer = tf.train.AdamOptimizer(learning_rate=0.002)
+    train_op = adam_optimizer.minimizer(avg_cost)
```
Comment (jacquesqiao) [inline tensorflow/understand_sentiment_lstm.py]: minimizer  => minimize?

Comment (dzhwinter) [reply tensorflow/understand_sentiment_lstm.py]: Done.

File: tensorflow/understand_sentiment_lstm.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+        lstm_input = tf.nn.embedding_lookup(embedding, data)
+
+        lstm_input = tf.unstack(lstm_input, seq_len, 1)
+        lstm_cell = tf.nn.rnn_cell.LSTMCell(
+            num_units=emb_dim, use_peepholes=False)
+
+        initial_state = lstm_cell.zero_state(batch_size, dtype=tf.float32)
+        outputs, states = rnn.static_rnn(
+            lstm_cell, lstm_input, dtype=tf.float32)
```
Comment (qingqing01) [inline tensorflow/understand_sentiment_lstm.py]: 这里看着只有一层lstm，可以设置多层stacked的lstm吗？

File: tensorflow/understand_sentiment_lstm.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+def run_benchmark(model):
+    start_time = time.time()
+
+    word_dict = paddle.dataset.imdb.word_dict()
+    print("load word dict successfully")
+    dict_dim = len(word_dict)
+    train_reader = paddle.batch(
+        paddle.reader.shuffle(
+            paddle.dataset.imdb.train(word_dict),
+            buf_size=FLAGS.batch_size * 10),
```
Comment (qingqing01) [inline tensorflow/understand_sentiment_lstm.py]: traing set总共25 000，buf_size设置成25000吧，增加buf_size，应该可以加速度reader。

Comment (dzhwinter) [reply tensorflow/understand_sentiment_lstm.py]: I don't think there is any difference, and the factor caused by buffer should be removed in our validation scripts.

File: tensorflow/understand_sentiment_lstm.py (hunk 2 above)
Comment (dzhwinter) [reply tensorflow/understand_sentiment_lstm.py]: Done.

**Pull request merged**